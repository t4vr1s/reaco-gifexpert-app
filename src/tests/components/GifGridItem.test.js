import { shallow } from 'enzyme'
import React from 'react'
import GifGridItem from '../../components/GifGridItem'

/* eslint-disable */
describe('Puebas al componente <GifGridItem />', () => {
  const url = 'https://localhost/img.png'
  const title = 'texto del titulo para la imagen de las pruebas'
  const wrapper = shallow(<GifGridItem url={url} title={title} />)

  test('Snapshot del componente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('debe tener un parrafo con el titulo', () => {
    const parrafo = wrapper.find('p')
    expect(parrafo.text().trim()).toBe(title)
  })

  test('debe tener en la imagen la url y el alt igual al de los props', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('debe tener la clase animate__fadeIn', () => {
    const div = wrapper.find('div')
    expect(div.prop('className').includes('animate__fadeIn')).toBe(true)
  })
})
