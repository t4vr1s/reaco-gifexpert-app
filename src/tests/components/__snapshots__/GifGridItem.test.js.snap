// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas al componente <GifGridItem /> Snapshot del componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="texto del titulo para la imagen de las pruebas"
    src="https://localhost/img.png"
  />
  <p>
    texto del titulo para la imagen de las pruebas
  </p>
</div>
`;
